src/libsync/propagatedownload.cpp

```diff
@@ -457,13 +443,65 @@ void PropagateDownloadFile::startDownload()
         propagator()->_journal->commit("download file start");
     }
 
+    if (isZsyncPropagationEnabled(propagator(), _item)) {
+        if (_item->_previousSize) {
+            // Retrieve zsync metadata file from the server
+            qCInfo(lcZsyncGet) << "Retrieving zsync metadaa file for:" << propagator()->getFilePath(_item->_file);
+            QNetworkRequest req;
+            req.setPriority(QNetworkRequest::LowPriority);
+
+            QUrl zsyncUrl = Utility::concatUrlPath(propagator()->account()->zsyncUrl(),
+                                                   propagator()->_remoteFolder + _item->_file);
+            auto job = propagator()->account()->sendRequest("GET", zsyncUrl, req);
+            connect(job, &SimpleNetworkJob::finishedSignal, this, &PropagateDownloadFile::slotZsyncGetMetaFinished);
+            return;
+        }
+        qCInfo(lcZsyncGet) << "No local copy of:" << propagator()->getFilePath(_item->_file);
+    }
+
+    startDownloadNormal();
+}
+
+
+void PropagateDownloadFile::slotZsyncGetMetaFinished(QNetworkReply *reply)
+{
+    int httpStatusCode = reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+    if (httpStatusCode != 200) {
+        qCInfo(lcPropagateDownload)
+            << "No remote zsync file for:"
```

Comment:
That definitely sounds like a better approach, but let's at least test the code as-is and implement this in the future?

Reply:
+1 to @guruz approach.

And honestly I would vote for fixing this before going into production. It might have also an impact on the server implementation that should not be postponed.

Reply:
To be honest about the @guruz approach, it's definitely nicer, but I'm not spending another two weeks on this PR for $1k. If you guys don't want to merge, then so be it.

Reply:
ok well it looks like adding the property is pretty easy, doing ...

Reply:
Done.